client/src/components/pages/Players.js

```diff
@@ -1,289 +1,339 @@
+        break;
+      default:
+        message.error("Failed to sort players");
+    }
+
+    return playersData;
+  };
 
-      return { players };
-    });
-  }
-
-  sortTeams(sort) {
-    this.setState((state) => {
-      const teams = [...state.teams];
-
-      if (sort === "alpha") {
-        teams.sort((x, y) => (x.name.toLowerCase() < y.name.toLowerCase() ? -1 : 1));
-      } else if (sort === "seed") {
-        teams.sort((x, y) => (x.seedNum || 0) - (y.seedNum || 0));
-      } else if (sort === "group") {
-        teams.sort((x, y) => ((x.group || "_") < (y.group || "_") ? -1 : 1));
-      } else if (sort === "rank") {
-        teams.sort(
+  const sortTeams = (sortMethod, unsortedTeams) => {
+    let teamsData;
+
+    switch (sortMethod) {
+      case "alpha":
+        teamsData = unsortedTeams.sort((x, y) =>
+          x.name.toLowerCase() < y.name.toLowerCase() ? -1 : 1
+        );
+        break;
+      case "seed":
+        teamsData = unsortedTeams.sort((x, y) => (x.seedNum || 0) - (y.seedNum || 0));
+        break;
+      case "group":
+        teamsData = unsortedTeams.sort((x, y) => ((x.group || "_") < (y.group || "_") ? -1 : 1));
+        break;
+      case "rank":
+        teamsData = unsortedTeams.sort(
           (x, y) =>
             x.players.reduce((sum, p) => sum + p.rank, 0) / x.players.length -
             y.players.reduce((sum, p) => sum + p.rank, 0) / y.players.length
         );
-      }
+        break;
+      default:
+        return message.error("Failed to sort teams");
+    }
+
+    return teamsData;
+  };
 
-      return { teams };
-    });
-  }
+  const handleDelete = async (username) => {
+    try {
+      await delet("/api/player", { tourney: tourney, username });
 
-  handleDelete = async (username) => {
-    await delet("/api/player", { tourney: this.props.tourney, username });
-    this.setState((state) => ({
-      players: state.players.filter((p) => p.username !== username),
-    }));
+      setPlayers(players.filter((p) => p.username !== username));
+      message.success("Player deleted!");
+    } catch {
+      message.error("Something went wrong, failed to delete player.");
+    }
   };
 
-  handleTeamDelete = async (_id) => {
-    await delet("/api/team", { tourney: this.props.tourney, _id });
-    this.setState((state) => ({
-      teams: state.teams.filter((p) => p._id !== _id),
-    }));
+  const handleTeamDelete = async (_id) => {
+    try {
+      await delet("/api/team", { tourney: tourney, _id });
+
+      setTeams(teams.filter((p) => p._id !== _id));
+    } catch {
+      message.error("Something went wrong, failed to delete team.");
+    }
   };
 
-  handleModeChange = (e) => {
+  const handleModeChange = (e) => {
     if (e.key === "players") {
-      this.setState(
-        {
-          mode: "players",
-          sort: "rank",
-        },
-        () => this.sortPlayers("rank")
-      );
+      setMode("players");
+      setSort("rank");
+      sortPlayers("rank", players);
```

Comment:
since `sortPlayers` just returns a sorted copy of the array, this needs to be:
```
setPlayers(sortPlayers("rank", players))
```
same with all the other calls to `sortPlayers` and `sortTeams`

or maybe even `sortedPlayers` is a better name to indicate that it's just returning a sorted copy rather than sorting it in place

Reply:
Noted